### PR TITLE
Update example notebooks

### PR DESCRIPTION
**Pull request by shuds13**
### Updating notebooks

Note that some links may be broken until this is pulled into main. 
Similar for some components of notebooks. 

### To try out notebooks in Colab while on this branch

To test forces and APOSMM notebooks on this branch in Colab you  need to pip install from this branch instead of just `pip install libensemble` do :
`pip install -U git+https://github.com/Libensemble/libensemble.git@examples/update_notebooks`

For forces notebook, also need to change URL to obtain forces.c while on this branch.
url = "https://raw.githubusercontent.com/Libensemble/libensemble/examples/update_notebookslibensemble/tests/scaling_tests/forces/forces_app/forces.c"

### Final check-list

- [x] Final run through notebooks on Colab 
- [x] When pull in, update Colab URLs for develop branch (and later main).
      - README and 3 tutorials in docs need updated URLs.

- [x]  Agree name change: *Executor with Electrostatic Forces* to "Ensemble with an MPI Application*
       We should name based on what user wants to do.

### Folllow up

#1337 adds another notebook with a more simple optimization example (multif-fiedlty with Ax), but  as there are some points to address, we should make that a follow up PR.

### Checklist

- [x]   Get working with latest libEnsemble and ensure using user functions in notebook:
  - [x]  sine
  - [x]  executor
  - [x]  aposmm

- [x]  Note spawn limitations in notebooks
- [x]  APOSMM - fix images - used general links to online image in repo
- [x] Update to match online tutorials (e.g., class interface).
- [x] Add links to docs/tutorials to view notebooks in colab.
- [x] Add back metadata necessary to prevent no runtime message in colab.
- [x] Should we add colab links in README/quickstart?
- [x] Why plot2N stopped showing in docs
- [x] Add forces example using an input file (tutorial and notebook)
- [x] For APOSMM, check minima found (versus check values in notebook/tutorial) are acceptable (see below).
- [x] We need to look at output of test_persistent_aposmm_scipy.py regression test.

APOSMM - fix three_hump_camel_func error
- [x] in notebook
- [x] in  tutorial

Questions:

- should we change to class interface - doing this at least to match online tutorials.
- should we update on binder - answer: moving to colab.
- Note: H, persis_info, flag are returned by ensemble.run() - but not used in our examples/regression tests. Instead we use ensemble.H etc.... This is apparent in notebook, as if I just use ensemble.run() it prints all the output.
- An issue this has high-lighted is that when using OO interface gen_specs (and others) can be defined as an object in the calling script, but becomes a dictionary in the user function. Perhaps it's better to keep these as dictionaries in our examples?

### CODE REVIEW

**Comment by shuds13** [comment]
APOSMM old minima (originally did 2000 sims) shown in tutorial and notebook.

[[ 0.08993295 -0.71265804]
 [ 1.70360676 -0.79614982]
 [-1.70368421  0.79606073]
 [-0.08988064  0.71270945]
 [-1.60699361 -0.56859108]
 [ 1.60713962  0.56869567]]

Now, in notebook, get:

[[ 0.08978611 -0.71258202]
 [-1.70380373  0.79614422]
 [-0.08985348  0.71271365]
 [ 1.70352649 -0.79610854]
 [ 1.60712922  0.56868726]
 [-1.60711706 -0.56868708]
 [ 1.70361564 -0.7960268 ]]

Now, in notebook, with sim_max 2000, get:

[[ 0.08978611 -0.71258202]
 [-1.70380373  0.79614422]
 [-0.08985348  0.71271365]
 [ 1.70352649 -0.79610854]
 [ 1.60712922  0.56868726]
 [-1.60711706 -0.56868708]
 [ 1.70361564 -0.7960268 ]]

File: examples/tutorials/aposmm/aposmm_tutorial_notebook.ipynb [hunk 1]
```diff
@@ -94,7 +110,7 @@
    "source": [
     "import numpy as np\n",
     "\n",
-    "from tutorial_six_hump_camel import six_hump_camel\n",
+    "# from tutorial_six_hump_camel import six_hump_camel\n",
```
Comment (jmlarson1) [inline examples/tutorials/aposmm/aposmm_tutorial_notebook.ipynb:113]: Should this line be removed from the tutorial?

File: examples/tutorials/simple_sine/sine_tutorial_notebook.ipynb [hunk 2]
```diff
@@ -205,51 +154,38 @@
...
-    "    },\n",
-    "}\n",
+    "import numpy as np\n",
     "\n",
-    "sim_specs = {\n",
-    "    \"sim_f\": sim_find_sine,  # Our simulator function\n",
-    "    \"in\": [\"x\"],  # Input field names. 'x' from gen_f output\n",
-    "    \"out\": [(\"y\", float)],\n",
-    "}  # sim_f output. 'y' = sine('x')\n",
+    "# from sine_gen import gen_random_sample\n",
```
Comment (jmlarson1) [inline examples/tutorials/simple_sine/sine_tutorial_notebook.ipynb:159]: DO we want these in the tutorial?

File: libensemble/tests/scaling_tests/forces/forces_simple_with_input_file/forces_simf.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    with open(input_file, "w") as f:
+        f.write(template.render(input_values))
+
+
+def run_forces(H, persis_info, sim_specs, libE_info):
+    """Runs the forces MPI application reading input from file"""
+
+    calc_status = 0
+
+    input_file = sim_specs["user"]["input_filename"]
```
Comment (jmlarson1) [inline libensemble/tests/scaling_tests/forces/forces_simple_with_input_file/forces_simf.py:34]: This variable (`input_file`) is set but not used in this function.

**Review by jlnav — APPROVED** [review]
Just ran through all the tutorial notebooks, very impressive! I agree with the various name-changes.

File: libensemble/tests/scaling_tests/forces/forces_simple_with_input_file/forces_simf.py (hunk 3 above)
Comment (jmlarson1) [reply libensemble/tests/scaling_tests/forces/forces_simple_with_input_file/forces_simf.py:34]: This is resolved in https://github.com/Libensemble/libensemble/pull/1338. I approve this PR

**Pull request merged**